src/track/serato/markers2.cpp

```diff
@@ -372,6 +372,11 @@ bool SeratoMarkers2::parse(SeratoMarkers2* seratoMarkers2, const QByteArray& out
 
 QByteArray SeratoMarkers2::dump() const {
     QByteArray data;
+    if (isEmpty() && getAllocatedSize() == 0) {
```

Comment:
How does getAllocated() size affect _emptyness_? If the marker is empty then allocating space just for padding bytes shouldn't be necessary. Or are there hidden contents?

Reply:
To reduce disk fragmentation, Serato pre-allocates at least 470 bytes for these tags. The tags contain a version number, base64-encoded marker entries and null bytes. So theoretical it would be possible to have a valid tag with version number and null bytes but without any markers inside.  The allocated size is set after successfully parsing the tag, so if the tag is valid but does not contain entries we shouldn't delete the tag content IMHO.

Reply:
I'm not sure if this is obvious to someone who stumbles over this code in the future. Just paste your explanation as a comment into the code, so it doesn't get lost in this PR. I think this is valuable information.